python/tests/res/fm/test_ecl_config.py

```diff
@@ -0,0 +1,120 @@
+
+                os.chmod(fname, stat.S_IEXEC)
+
+            intel_path = "intel"
+            os.environ["ENV1"] = "A"
+            os.environ["ENV2"] = "C"
+            d = {"env" : {"LICENSE_SERVER" : "license@company.com"},
+                 "simulators": {"ecl100" : {"2015" : {"scalar": {"executable" : scalar_exe},
+                                                      "mpi"   : {"executable" : mpi_exe,
+                                                                 "mpirun"     : mpi_run,
+                                                                 "env" : {"I_MPI_ROOT" : "$ENV1:B:$ENV2",
+                                                                          "P4_RSHCOMMAND" : "",
+                                                                          "LD_LIBRARY_PATH" : "{}:$LD_LIBRARY_PATH".format(intel_path),
+                                                                          "PATH" : "{}/bin64:$PATH".format(intel_path)}}}}
+                                ,
+                                "flow" : {"2018.04" : {"scalar" : {"executable" : "/does/not/exist"},
+                                                       "mpi" : {"executable" : mpi_exe,
+                                                                "mpirun"  : "/does/not/exist"}}}}}
+
+            with open("file.yml", "w") as f:
+                f.write( yaml.dump(d) )
+
+            conf = EclConfig()
+            # Fails because there is no simulator ecl99
+            with self.assertRaises(KeyError):
+                sim = conf.sim("ecl99", "2015")
+
+            # Fails because there is no version 2020 for ecl100
+            with self.assertRaises(KeyError):
+                sim = conf.sim("ecl100", "2020")
+
+            # Fails because the 2018.04 version of flow points to a not existing executable
+            with self.assertRaises(OSError):
+                sim = conf.sim("flow", "2018.04")
+
+            # Fails because the 2018.04 mpir version points to a non existing mpirun binary
+            with self.assertRaises(OSError):
+                sim = conf.mpi_sim("flow", "2018.04")
+
+            self.assertFalse( conf.has_sim("flowiX", "2018.04"))
+            self.assertFalse( conf.has_mpi_sim("flow", "2018.04"))
+
+            mpi_sim = conf.mpi_sim("ecl100", "2015")
+            # Check that global environment has been propagated down.
+            self.assertIn("LICENSE_SERVER", mpi_sim.env)
+
+            # Check replacement of $ENV_VAR in values.
+            self.assertEqual(mpi_sim.env["I_MPI_ROOT"], "A:B:C")
+            self.assertEqual(len(mpi_sim.env), 1 + 4)
+
+            sim = conf.sim("ecl100", "2015")
+            self.assertEqual(sim.executable, scalar_exe)
+            self.assertIsNone(sim.mpirun)
+
+            with self.assertRaises(Exception):
+                simulators = conf.simulators()
+
+            simulators = conf.simulators(strict = False)
+            self.assertEqual(len(simulators), 2)
```

Comment:
This test is a bit "heavy". It would have been maybe clearer if we had some test data in some test folder instead of generating it on the fly. And if different functionalities were tested in different test functions. 
But I'd say that for now this is enough, this is going to be one of the most used forward models, possible bugs would pop up rather quickly anyway